### PR TITLE
scripts: use pathlib and avoid page iteration

### PR DESCRIPTION
**Pull request by vitorhcl**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).

Now set-more-info-link.py uses pathlib instead of os.path, which gave 2 clear benefits:
- Simplified the code
- It finally works properly on Windows (it wasn't getting the TLDR_ROOT automatically because of backslash, so I imported update-link.py's get_tldr_root())

Also, when a page is given via --page/-p, it doesn't iterate over each page on each platform anymore.

### CODE REVIEW

File: scripts/set-more-info-link.py [hunk 1]
```diff
@@ -90,22 +91,20 @@
...
+def get_tldr_root() -> Path:
+    f = Path(__file__).resolve()
+    return next(path for path in f.parents if path.name == "tldr")
 
     if "TLDR_ROOT" in os.environ:
-        return os.environ["TLDR_ROOT"]
+        return Path(os.environ["TLDR_ROOT"])
     print(
-        "\x1b[31mPlease set TLDR_ROOT to the location of a clone of https://github.com/tldr-pages/tldr."
+        "Please set TLDR_ROOT to the location of a clone of https://github.com/tldr-pages/tldr."
```
[the remark this answers is not in the document]
Comment (vitorhcl) [reply scripts/set-more-info-link.py:101]: Oh, I just copied from the other script, the other script had a logger that prefixed all errors, warnings, etc. accordingly, so I just replaced the logger.error, with a print. But sure, we can put the color again.

**Review by kbdharun — APPROVED** [review]
LGTM, Thanks for your contribution.

File: scripts/set-more-info-link.py [hunk 2]
```diff
@@ -239,59 +238,48 @@ def main():
...
-    rel_paths = []
+    target_paths = []
 
     # Use '--page' option
     if args.page != "":
-        target_paths = []
-
-        if not args.page.lower().endswith(".md"):
-            args.page = f"{args.page}.md"
+        args.page = f"{args.page.rstrip('.md')}.md"
```
Comment (blueskyson) [inline scripts/set-more-info-link.py:247]: I think `rstrip` is misused here. I found some error cases:
```python
>>> print('md5sum'.rstrip('.md'))
md5su
>>> print('md5sum.md'.rstrip('.md'))
md5su
>>> print('dd'.rstrip('.md'))

>>> print('dd.md'.rstrip('.md'))

```

**Review by blueskyson — CHANGES_REQUESTED** [review]

**Review by blueskyson — APPROVED** [review]
Look good to me. This can be merged after someone tests single-page and sync-all-pages functionality.

**Review by kbdharun — APPROVED** [review]
> Look good to me. This can be merged after someone tests single-page and sync-all-pages functionality.

Tested both functionalities locally on Windows. It works fine, the PR is GTG.

**Pull request merged**